id_manager/meson.build

```diff
@@ -0,0 +1,70 @@
+project('id_manager', 'cpp',
+  version : '0.0.0',
+  license : 'MPLv2')
+
+project_description = 'A library to manage arbitrary kind of ids within your application.'
+
+uuid = dependency('uuid')
+
+project_headers = [
+  'include/IdentifierConstPtr.h',  
+  'include/Identifier.h',  
+  'include/IDManager.h',
+  'include/WildcardID.h'
+]
+
+project_source_files = [
+  'src/Identifier.cpp',
+  'src/IDManager.cpp',
+  'src/WildcardID.cpp'
+]
+
+project_test_files = [
+  'tests/src/IDTests.cpp'
+]
+
+build_args = [
+]
+
+
+# ===================================================================
+
+# ======
+# Target
+# ======
+
+public_headers = include_directories('include')
+
+build_args += [
+  '-DPROJECT_NAME=' + meson.project_name(),
+  '-DPROJECT_VERSION=' + meson.project_version(),
+]
+
+project_target = library(
+  meson.project_name(),
+  project_source_files,
+  c_args : build_args,
+  include_directories : public_headers,
+  dependencies : uuid
+)
+
+
+# =======
+# Project
+# =======
+
+# Make this library usable as a Meson subproject.
+project_dep = declare_dependency(
+  include_directories: public_headers,
+  link_with : project_target,
+  version : meson.project_version(),
+)
+set_variable(meson.project_name() + '_dep', project_dep)
```

Comment:
set_variable is a weird API, why don't you just create the variable with the correct name by default?

Reply:
I read about meson.overwrite_dependency. I think that way, we don't need the declare_dependency variable to be uniformly named.

Reply:
Yeah, override_dependency is nice.

You may want to add this to the top `project()` function:

```meson
    meson_version: '>=0.54.0',
```

So that people trying to build the project with older versions of Meson get told which version they need.

Alternatively, you can support older versions of Meson like this:

```meson
if meson.version().version_compare('>=0.54.0')
     meson.override_dependency(meson.project_name(), project_dep)
endif
```